### ISSUE TITLE
[New Tracker REQ] TNT Village

### ISSUE THREAD

**Issue by EffeF** [Jackett/Jackett#1111]
**Indexer Name:**
TNT Village

**Type:**
Movies, Music, TV Movies and Shows, documentaries, Students Releases, E-Books, Linux, Anime, Cartoons, Macintosh, Windows Software, Pc Game, Playstation, Music Videos, Sports, Theatre, Wrestling, Other, Xbox, Images backgrounds, Other games, TV series, Fumetteria, Trash, Nintendo, A-Book, Podcast, Kiosk, Mobile

**Indexer URL:**
http://www.tntvillage.scambioetico.org/?releaselist

**Indexer Language:**
Italian

**Invite:**
Unnecessary

**Comment by bonny1992** [comment]
I am currently working on this tracker, but I'm encountering some problems.

It seems that the post request needed for the search function is not sent correctly.
It requires 3 values:
- **cat** (value: 0)
- **page** (value: 1)
- **srcrel** (value: our search terms)

Trying them with Postman returns the correct results, but not with Jackett (which returns the page like no request has been made)

I tried with some different combinations of the search function, but with no luck at the moment.
```  
 search:
    path: /src/releaselist.php
    method: "post"
    inputs:
      $raw: "cat=0&page=1&srcrel={{ .Query.Keywords }}"
```

or

```  
 search:
    path: /src/releaselist.php
    method: post
    inputs:
      cat: 0
      page: 1
      srcrel: "{{ .Query.Keywords }}"
```

(I tried _"pos"_ and _post_, or _form_ and _"form"_)

Here's the resulting URL that the advanced logging of Jackett returns:
http://www.tntvillage.scambioetico.org/src/releaselist.php?cat=0&page=1&srcrel=test
which is a correct representation of the request, but it isn't working.

Any advice?

**Comment by kaso17** [comment]
post support is simply not implemented yet, no tacker needed it. I'll add it soon.

**Comment by bonny1992** [comment]
Oh.

I supposed it was working because of the logins, some of them requires the post request, after all.

**Comment by kaso17** [comment]
For login it's supported since the beginning.
For search there was never the need.
It's supported now with 3c46daf7eefbdbafbc6465267605c07c367db578
Please try pre releae v0.7.1150
You'll have to use the newer multi path syntax:

```yaml
  search:
    paths:
      - path: /src/releaselist.php
        method: post
    inputs:
      cat: 0
      page: 1
      srcrel: "{{ .Keywords }}"
```

**Comment by bonny1992** [comment]
@kaso17 
Ok, now with pre release it works like a charm!
I have two problems, yet.

On the releaselist page it does not show the publish date nor the torrent size. Is there any way to navigate to the details page and fetch them from there? Apart from that it works like a charm:


![JacketTNTVillage](http://image.prntscr.com/image/0e2c7bba06e04197b8909dddf0f08c4e.png "TNTVillage tracker on Jackett - Manual search")
![TNTNative](http://image.prntscr.com/image/4a42af28a20349738081aff08e17e533.png "TNTVillage tracker on native page")

**Comment by kaso17** [comment]
No, accessing any other page than the search results page is not supported. I wouldn't recommend it either due to the additional site requests involved (causing a lot of extra load on the server and the corresponding extra delay to get the search result pages).

**Comment by bonny1992** [comment]
So what we do? Do we omit the date and size?

**Comment by kaso17** [comment]
yes

**Comment by bonny1992** [comment]
Perfect.

Then here you go the yml file:
http://pastebin.com/tHKn7i4F

**Comment by kaso17** [comment]
added with 38f78e116cb604013b6ef69097822f36e721f585 (added movie search and set downloadvolume factor to 0), thank you

**Comment by rafuz** [comment]
Sorry but setting size to 0 breaks Sonarr results parsing.
It needs to know file size to compare quality categories.

**Comment by kaso17** [comment]
In that case I suggest setting it to an arbitrary value, e.g.:
```yaml
size:
  text: "1G"
```
I also suggest to contact the TNT village admins and ask them to add the size to the search results.

**Comment by rafuz** [comment]
> I also suggest to contact the TNT village admins and ask them to add the size to the search results.

This is a good idea
In the meantime I will try adding a fixed size value to see if Sonarr parser is able to get results

**Comment by rafuz** [comment]
This is the result when using the search from Couchpotato :-(

`HandleCore(): unhandled exception: Message = Nullable object must have a value.
Data = System.Collections.ListDictionaryInternal
InnerException = 
TargetSite = Void Throw()
StackTrace =   at System.Nullable`1[T].get_Value () [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at Jackett.Controllers.PotatoController+<Call>c__async0.MoveNext () [0x0067c] in C:\projects\jackett\src\Jackett\Controllers\PotatoController.cs:148 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Threading.Tasks.TaskHelpersExtensions+<CastToObject>d__3`1[T].MoveNext () [0x00063] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Tracing.ITraceWriterExtensions+<TraceBeginEndAsyncCore>d__18`1[TResult].MoveNext () [0x000e5] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Controllers.ApiControllerActionInvoker+<InvokeActionAsyncCore>d__0.MoveNext () [0x000c7] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Tracing.ITraceWriterExtensions+<TraceBeginEndAsyncCore>d__18`1[TResult].MoveNext () [0x000e5] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Filters.ActionFilterAttribute+<CallOnActionExecutedAsync>d__5.MoveNext () [0x0009d] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Filters.ActionFilterAttribute+<CallOnActionExecutedAsync>d__5.MoveNext () [0x001f6] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Filters.ActionFilterAttribute+<ExecuteActionFilterAsyncCore>d__0.MoveNext () [0x0011b] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Controllers.ActionFilterResult+<ExecuteAsync>d__2.MoveNext () [0x001f1] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Tracing.Tracers.HttpControllerTracer+<ExecuteAsyncCore>d__5.MoveNext () [0x0008b] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Tracing.ITraceWriterExtensions+<TraceBeginEndAsyncCore>d__18`1[TResult].MoveNext () [0x000e5] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Dispatcher.HttpControllerDispatcher+<SendAsync>d__1.MoveNext () [0x0018d] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Dispatcher.HttpControllerDispatcher+<SendAsync>d__1.MoveNext () [0x0030d] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.Tracing.ITraceWriterExtensions+<TraceBeginEndAsyncCore>d__18`1[TResult].MoveNext () [0x000e5] in <f99f496cb0d249c1a945c1fcabce1695>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Web.Http.HttpServer+<SendAsync>d__0.MoveNext () [0x00187] in <f99f496cb0d249c1a945c1fcabce1695>:0 
HelpLink = 
Source = mscorlib
HResult = -2146233079`

**Comment by kaso17** [comment]
Ok, size is required for the potato feed :)
Should work with f2a03324f3872a7ca6b45c7fbadfdb048bfb6c8e

**Comment by rafuz** [comment]
5G is read as 5 byte.
I am using "3000000000" to set it to 3 GByte

It could be nice to report different default sizes based on category (1 GByte for TV and Anime and 2 GB for Movies)

Anyway the real solution is to get the real size from the search.

**Comment by kaso17** [comment]
For "dynamic" sizes something like this should work:
```yaml
      size:
        selector: td:nth-child(3) a
        case:
          a[href*="&cat=4"]: "5GB"
          a[href*="&cat=2"]: "100MB"
          "*": "1GB"
```

please create a pull request for it.

If the admins say no to adding the size I'll add a way to grab it from the details page.

**Comment by rafuz** [comment]
I have already asked, waiting for an answer.

**Comment by bonny1992** [comment]
It is extremely unlikely that the admins will say yes.
They are a lot against the use of tools like Sickrage or Couchpotato.

**Comment by rafuz** [comment]
TLDR: the admins says no

> It is extremely unlikely that the admins will say yes.
They are a lot against the use of tools like Sickrage or Couchpotato.

This is true but I did not asked it for download automation, but just to improve the search feature.
Anyway they answered that the page we are using for search is not meant to get all the details from a torrent. I should use the RSS sync or login to the full website to get all details.

**Comment by bonny1992** [comment]
@rafuz 
It is not needed to login, once you have the release detail link (which is a thing we already have) you can navigate into it with ease.

So we'll need to wait for @kaso17 if he has the time and will to add a way to grab the detail of the torrent from the detail page.
In that way I could even add the correct posting date.

**Comment by kaso17** [comment]
Is the size/time included in the results if you're logged in? Then that would be the simple solution.

**Comment by bonny1992** [comment]
Do you mean in the details page or in the releaselist? 
In the releaselist it doesn't change if you're logged or not, and in the details page you can only thank the uploader of reply to the topic if logged. 
The date/size is accessible without logging.

**Comment by rafuz** [comment]
From the release list if you click on an item you will open the release details:

![image](https://cloud.githubusercontent.com/assets/22271829/23939497/362dbb2e-0961-11e7-8373-7e500799d376.png)

You can find the torrent info at the bottom of the first post, I think that something like that has been done in a couchpotato plugin: [https://github.com/autoscatto/tntvillage-cp-provider](url)

**Comment by kaso17** [comment]
The CP plugin is using the "dim" column for size from http://forum.tntvillage.scambioetico.org/index.php?act=allreleases, which doesn't make any sense to me.
It additionally queries the details page, but just to get the magnet link...

**Comment by kaso17** [comment]
Will add a feature to query other pages.

**Comment by garfield69** [comment]
![image](https://user-images.githubusercontent.com/6933240/64050109-26b97a00-cbcb-11e9-8126-9fcafc8e7de5.png)

**Comment by jokerigno** [comment]
Up. 

TNT is up and running again. @garfield69 can be readded to jackett?

http://tntvillage.scambioetico.org/?releaselist

**Comment by garfield69** [comment]
See #7926
